dbos/_sys_db.py

```diff
@@ -1112,88 +1113,153 @@ def start_queued_workflows(self, queue: "Queue", executor_id: str) -> List[str]:
             # Execute with snapshot isolation to ensure multiple workers respect limits
             c.execute(sa.text("SET TRANSACTION ISOLATION LEVEL REPEATABLE READ"))
 
-            # If there is a limiter, compute how many functions have started in its period.
-            if queue.limiter is not None:
-                query = (
-                    sa.select(sa.func.count())
-                    .select_from(SystemSchema.workflow_queue)
-                    .where(SystemSchema.workflow_queue.c.queue_name == queue.name)
-                    .where(
-                        SystemSchema.workflow_queue.c.started_at_epoch_ms.isnot(None)
+            ret_ids: list[str] = []
+            try:
```

Comment:
We use a try/finally to ensure the garbage collection always happens.

The new logic otherwise relies on returning early when now tasks are eligible for dequeue for this worker.